go/libraries/doltcore/sqle/commit_diff_table.go

```diff
@@ -0,0 +1,397 @@
+	}
+
+	for _, ref := range refs {
+		cm, err := ddb.ResolveRef(ctx, ref)
+
+		if err != nil {
+			return nil, err
+		}
+
+		cmRoot, err := cm.GetRootValue()
+
+		if err != nil {
+			return nil, err
+		}
+
+		ss, found, err = cmRoot.GetSuperSchema(ctx, tblName)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if found {
+			superSchemas = append(superSchemas, ss)
+		}
+	}
+
+	if len(superSchemas) == 0 {
+		return nil, sql.ErrTableNotFound.New(tblName)
+	}
+
+	superDuperSchema, err := schema.SuperSchemaUnion(superSchemas...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return superDuperSchema, nil
+}
+
+func (dt *CommitDiffTable) Name() string {
+	return doltdb.DoltCommitDiffTablePrefix + dt.name
+}
+
+func (dt *CommitDiffTable) String() string {
+	return doltdb.DoltCommitDiffTablePrefix + dt.name
+}
+
+func (dt *CommitDiffTable) Schema() sql.Schema {
+	return dt.sqlSch
+}
+
+type SliceOfPartitionsItr struct {
+	partitions []sql.Partition
+	i          int
+	mu         *sync.Mutex
+}
+
+func NewSliceOfPartitionsItr(partitions []sql.Partition) *SliceOfPartitionsItr {
+	return &SliceOfPartitionsItr{
+		partitions: partitions,
+		mu:         &sync.Mutex{},
+	}
+}
+
+func (itr *SliceOfPartitionsItr) Next() (sql.Partition, error) {
+	itr.mu.Lock()
+	defer itr.mu.Unlock()
+
+	if itr.i >= len(itr.partitions) {
+		return nil, io.EOF
+	}
+
+	next := itr.partitions[itr.i]
+	itr.i++
+
+	return next, nil
+}
+
+func (itr *SliceOfPartitionsItr) Close() error {
+	return nil
+}
+
+func (dt *CommitDiffTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error) {
+	if dt.requiredFilterErr != nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), dt.requiredFilterErr)
+	} else if dt.toCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneToCommit)
+	} else if dt.fromCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneFromCommit)
+	}
+
+	toRoot, toName, toDate, err := dt.rootValForFilter(ctx, dt.toCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromRoot, fromName, fromDate, err := dt.rootValForFilter(ctx, dt.fromCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toTable, _, err := toRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromTable, _, err := fromRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSliceOfPartitionsItr([]sql.Partition{diffPartition{
+		to:       toTable,
+		from:     fromTable,
+		toName:   toName,
+		fromName: fromName,
+		toDate:   toDate,
+		fromDate: fromDate,
+	}}), nil
+}
+
+func (dt *CommitDiffTable) rootValForFilter(ctx *sql.Context, eqFilter *expression.Equals) (*doltdb.RootValue, string, *types.Timestamp, error) {
+	var hashStr string
+	var ok bool
+	if _, isGetField := eqFilter.BinaryExpression.Left.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Left.Eval(ctx, nil)
```

Comment:
This wont' work for all expressions (like a subselect, for example). Should do something reasonable in that case.

Reply:
The row can't be nil I mean

Reply:
You probably need to do an inspection at set time and only allow certain types of expressions